_posts/2017-04-24-solarized.md

```diff
@@ -0,0 +1,545 @@
+```
+
+This allows me to use Solarized with True colors (24-bit colors, or 16 million
+colors, supported by Termite). It also adds two interesting key bindings (my
+Leader key being <kbd>,</kbd>):
+
+* <kbd>,</kbd>-<kbd>B</kbd> will toggle the dark and light modes in Vim
+* <kbd>,</kbd>-<kbd>-</kbd> and <kbd>,</kbd>-<kbd>+</kbd> allow me to change the
+  contrast of the selected mode.
+
+**Note:** I'm using [vim-plug](https://github.com/junegunn/vim-plug) as Vim
+plugin manager, which is simple to use and full of smart features, for example
+allowing to lazy-load modules when their command is called.
+
+```vim
+pattern="colorscheme solarized8_"
+perl -e "s/${pattern}${oldmode}/${pattern}${newmode}/g" -pi ${HOME}.vimrc
+for socket in /tmp/nvimsocket*; do
+    nvr --servername $socket --remote-send ",B<Enter>";
+done;
+```
+
+I then used [nvr](https://github.com/mhinz/neovim-remote) to send a color
+toggle command to running vim sessions. The reason I used it is that the
+[+clientserver](https://neovim.io/doc/user/remote.html#clientserver) of neovim
+goes through the X-server (like [vim
+--remote](http://vimdoc.sourceforge.net/htmldoc/remote.html) for that matter),
+or I'm working with neovim in a terminal without GUI.
+To create new nvim servers everytime I open vim, I added to my `~/.zshrc` file
+the following:
+
+```zsh
+function vim () {
+    COUNT=$(($((ls /tmp/nvimsocket*) 2>&/dev/null|tail -n 1|grep -Po 'nvimsocket\K[0-9]*$') + 1))
+    printf -v COUNT "%05d" $COUNT
+    NVIM_LISTEN_ADDRESS=/tmp/nvimsocket${COUNT} nvim $@
+}
+```
+This function will create a new socket file suffixed with an incremented number,
+so that nvr can easily retrieve all server and send them update commands.
+
+# [](#gtk) GTK
+
+Since I'm using several GTK 2.0 applications, like [Gimp](https://www.gimp.org/),
+[Thunar](http://docs.xfce.org/xfce/thunar/start) or
+[Deluge](http://deluge-torrent.org/), it also makes sense to update
+the gtk 2.0 color theme.
+
+I went with the [Numix
+Solarized](https://github.com/Ferdi265/numix-solarized-gtk-theme) themes.
+
+To update opened applications, I re-used [cpoakes'
+script](http://crunchbang.org/forums/viewtopic.php?pid=428517#p428517) to send events.
+
+```console
+$ echo "include \"/usr/share/themes/NumixSolarized$([[ $newmode == 'dark' ]] \
+    && echo 'Dark')/gtk-2.0/gtkrc\"" > ${HOME}/.gtkrc-2.0
+$ python2.7 ${HOME}/scripts/gtkreload &
+```
+
+
+
+# [](#chrome)Google Chrome
+
+Google Chrome currently uses GTK 2.0, but will eventually use GTK 3.0 by
+default.
+Note that enabling the GTK+ theme in `Settings → Appearance` isn't enough to get
+a solarized look-and-feel. Google uses around 80 parameters for its design, and
+will only pick some colors from GTK, while using custom colors for other
+parameters.
+In order to truly customize the Chrome theme, it's require to build an
+extension-like theme, which can then be uploaded to the Chrome store and used
+from there.
```

Comment:
> +In order to truly customize the Chrome theme, it's require to build an
+extension-like theme, which can then be uploaded to the Chrome store and used
+from there.

To truly customise the Chrome theme, an extension-like theme needs to be built. It can then be uploaded to the Chrome store and used by anyone from there.